### ISSUE TITLE
Is there a way to set my own context vars that are visible in all handlers?

### ISSUE THREAD

**Issue by zoopp** [nolar/kopf#699]
## Question

I want to load some settings from the environment that I'm going to use later while the operator is running. Although not important, for settings management I use [pydantic](https://pydantic-docs.helpmanual.io/usage/settings/).

My initial thought was to make use of the startup handler to load these settings into a context var but it won't work because, I assume, the startup handler is run as a task that is not run in the same context as the other handlers so the change will consequently not be visible to other handlers.

e.g. this is what I had in mind:

```python
from pydantic import Base settings

class OperatorSettings(BaseSettings):
    class Config:
        env_file = ".env"
        env_prefix = "OPERATOR_"
    ...


settings: ContextVar[OperatorSettings] = ContextVar("settings")


@kopf.on.startup(errors=kopf.ErrorsMode.PERMANENT)
async def init_settings(logger: logging.Logger, **kwargs):
    logger.info("loading settings..")
    settings.set(OperatorSettings())
    logger.info("loaded the following settings: %r", settings)


@kopf.on.create(...):
async def create_fn(...):
    use_settings(settings.get())   # setting.get() should not raise a lookup error
```

I can work around this by not using contextvars and use `settings` as a global variable instead but, if possible, I would like to avoid it.

## Checklist

- [x] I have read the [documentation](https://kopf.readthedocs.io/en/latest/) and searched there for the problem
- [x] I have searched in the [GitHub Issues](https://github.com/nolar/kopf/issues?utf8=%E2%9C%93&q=) for similar questions

## Keywords

contextvars, startup, settings, configuration

**Comment by nolar** [comment]
Oh! That is a well-timed question! Several minutes ago, I’ve announced a new feature exactly for this purpose: https://twitter.com/nolar/status/1364872985757310976 — there is a picture with a code snippet. 

It is in a release candidate yet, but will be fully released fast. 

Docs: https://kopf.readthedocs.io/en/latest/memos/

Related/duplicate: #347, #367, implemented #667.

**Comment by zoopp** [comment]
Hah, the timing couldn't be better indeed :smile:. Thank you for the reply and the work you put into this project.

I just played a bit with it and it works well for this use case! From my point of view this question is answered! I'm in the relatively early stages of writing my operator so I'll take this chance to play a bit with the release candidate.
